benchmarks/battleship/benchmark/src/main/java/com/android/designcompose/benchmark/battleship/benchmark/StartupBenchmarks.kt

```diff
@@ -16,65 +16,38 @@
 
 package com.android.designcompose.benchmark.battleship.benchmark
 
-import androidx.benchmark.macro.CompilationMode
+import androidx.benchmark.macro.ExperimentalMetricApi
 import androidx.benchmark.macro.StartupMode
 import androidx.benchmark.macro.StartupTimingMetric
+import androidx.benchmark.macro.TraceSectionMetric
 import androidx.benchmark.macro.junit4.MacrobenchmarkRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.android.designcompose.DCTraces
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 
-const val PACKAGE_NAME = "com.android.designcompose.benchmarks.battleship"
-
 @RunWith(AndroidJUnit4::class)
 class StartupBenchmarks {
     @get:Rule val benchmarkRule = MacrobenchmarkRule()
 
+    @OptIn(ExperimentalMetricApi::class)
     @Test
     fun coldStartup() =
         benchmarkRule.measureRepeated(
             packageName = PACKAGE_NAME,
-            metrics = listOf(StartupTimingMetric()),
-            iterations = 5,
-            startupMode = StartupMode.COLD
-        ) {
-            pressHome()
-            startActivityAndWait()
-        }
-
-    @Test
-    fun hotStartup() =
-        benchmarkRule.measureRepeated(
-            packageName = PACKAGE_NAME,
-            metrics = listOf(StartupTimingMetric()),
-            iterations = 5,
-            startupMode = StartupMode.HOT
-        ) {
-            pressHome()
-            startActivityAndWait()
-        }
-
-    @Test
-    fun noPrecompileStartup() =
-        benchmarkRule.measureRepeated(
-            packageName = PACKAGE_NAME,
-            metrics = listOf(StartupTimingMetric()),
-            iterations = 5,
-            compilationMode = CompilationMode.None(),
-            startupMode = StartupMode.COLD
-        ) {
-            pressHome()
-            startActivityAndWait()
-        }
-
-    @Test
-    fun fullPrecompileStartup() =
-        benchmarkRule.measureRepeated(
-            packageName = PACKAGE_NAME,
-            metrics = listOf(StartupTimingMetric()),
+            metrics =
+                listOf(
+                    StartupTimingMetric(),
+                    TraceSectionMetric(DCTraces.DECODEDISKDOC),
+                    TraceSectionMetric(DCTraces.DESIGNDOCINTERNAL),
+                    TraceSectionMetric(
+                        DCTraces.DESIGNFRAME_DE_SUBSCRIBE,
+                        TraceSectionMetric.Mode.Sum
+                    ),
+                    TraceSectionMetric(DCTraces.FETCHDOCUMENTS)
+                ),
```

Comment:
What does this list do? Does it sum up all of these timings?

Reply:
That's described here: https://developer.android.com/topic/performance/benchmarking/macrobenchmark-metrics